### PR TITLE
Integrate datum-factory in smb-avro

### PR DESCRIPTION
**Pull request by RustedBones**

### CODE REVIEW

File: scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java [hunk 1]
```diff
@@ -103,50 +82,40 @@ public void populateDisplayData(Builder builder) {
...
-                      org.apache.beam.sdk.extensions.avro.schemas.utils.AvroUtils
-                          .addLogicalTypeConversions(data);
-                      return new ReflectDatumWriter<>(writer, data);
-                    });
+        ((AvroIO.Sink<ValueT>) AvroIO.sink(getSchema()))
+                .withDatumWriterFactory(datumFactory)
+                .withCodec(codec.getCodec());
 
     if (metadata != null) {
       return sink.withMetadata(metadata);
```
Comment (RustedBones) [inline scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java:97]: I think there was a bug here: codec was not set if metadata was not null

Comment (clairemcginty) [reply scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java:97]: ack! good thing there wasn't much adoption of the the metadata feature here... 😬

**Review by clairemcginty — APPROVED** [review]

File: scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java (hunk 1 above)
Comment (RustedBones) [reply scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java:97]: BTW, metadata is not exposed on the IO level. Is this expected ?

Comment (clairemcginty) [reply scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java:97]: it's not part of AvroSortedBucketIO? It should be 😬

**Pull request merged**